core/services/job/orm.go

```diff
@@ -391,7 +392,14 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 		}
 
 		jb.PipelineSpecID = pipelineSpecID
-		err = o.InsertJob(jb, pg.WithQueryer(tx))
+
+		// If provided job has an id, emplace it, otherwise insert it with a new id.
+		if jb.ID != 0 {
+			err = o.EmplaceJob(jb, pg.WithQueryer(tx))
+		} else {
+			err = o.InsertJob(jb, pg.WithQueryer(tx))
+		}
```

Comment:
I think this is unnecessary. You can probably just modify `InsertJob` to respect the passed ID